### PR TITLE
created payments table

### PR DESCRIPTION
**Pull request by roskzhu**
## Notion ticket link

[create payments table](https://www.notion.so/uwblueprintexecs/e2d9163a2f8c436bbb36172066753d40?v=9518aa1af04b4478b3d5e840f4ffcc7f&p=6b4d03cd50874339a33c8ad86da9fb61&pm=s)

## Implementation description
* defined schema for payments table
* created and applied migration

## Steps to test
1. NA

## What should reviewers focus on?
* Please review our schema columns :)

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by mat-ng** [comment]
Great work so far! I tested that the Payment table gets created successfully and I like the constraints.

I think the Donation table referencing the Payment table ultimately makes more sense. These changes would help massively 🙏

File: backend/typescript/prisma/migrations/20240526175607_create_payments_table/migration.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+-- This is an empty migration.
```
[the remark this answers is not in the document]
Comment (roskzhu) [reply backend/typescript/prisma/migrations/20240526175607_create_payments_table/migration.sql:1]: good question

File: backend/typescript/prisma/schema.prisma [hunk 2]
```diff
@@ -39,6 +39,7 @@ model Donation {
   cause_id                Int
   is_recurring            Recurrence
   confirmation_email_sent Boolean
+  transactions            Payment[]
```
[the remark this answers is not in the document]
Comment (roskzhu) [reply backend/typescript/prisma/schema.prisma]: i think this comment is referring to an old commit, should be updated now

**Comment by mat-ng** [comment]
```
de_ts_backend  |     Property 'payment_id' is missing in type '{ user_id: string; amount: number; donation_date: Date; cause_id: number; is_recurring: Recurrence; confirmation_email_sent: boolean; }' but required in type 'DonationUncheckedCreateInput'.
```

Thanks for making those changes to the `Donation` table! But a few more changes are going to be needed in the `Donation` service. For example, we'll have to pass in a `payment_id` parameter when we create a `Donation`. Are you guys able to fix those?

The errors can be seen when you try spinning up the Docker containers.

**Review by mat-ng — APPROVED** [review]

**Pull request merged**